pybuildkite/client.py

```diff
@@ -60,8 +89,10 @@ def request(
         if headers is None:
             raise ValueError("headers cannot be None")
 
-        if self.access_token:
+        if self.access_token and self._api_version == APIVersion.V2:
             headers["Authorization"] = "Bearer {}".format(self.access_token)
+        elif self.agent_token and self._api_version == APIVersion.V3:
+            headers["Authorization"] = "Token {}".format(self.agent_token)
```

Comment:
Will be cleaner if we separate the clients. Perhaps we can have a base client to keep some of the logic in this class DRY.